test-data/README.md

```diff
@@ -0,0 +1,5 @@
+# Test Data
```

Comment:
Again, should be  
[test_data_tree.txt](https://github.com/TileDB-Inc/TileDB-Vector-Search/files/13953357/test_data_tree.txt)

(Applies to all changes below.)

I can send you a tar file with all of this directory if it would be helpful.